erizo/src/erizo/rtp/SenderBandwidthEstimantionHandler.cpp

```diff
@@ -96,29 +99,54 @@ void SenderBandwidthEstimationHandler::read(Context *ctx, std::shared_ptr<dataPa
                 });
-            // TODO(pedro) Implement alternative when there are no REMBs
-            if (received_remb_ && value != sr_delay_data_.end()) {
+            // TODO(pedro) Maybe implement better alternative when there are no REMBs
+            if (value != sr_delay_data_.end()) {
                 uint32_t delay = now_ms - (*value)->sr_send_time - delay_since_last_ms;
                 ELOG_DEBUG("%s message: Updating Estimate with RR, fraction_lost: %u, "
                     "delay: %u, period_packets_sent_: %u",
                     connection_->toLog(), chead->getFractionLost(), delay, period_packets_sent_);
                 sender_bwe_->UpdateReceiverBlock(chead->getFractionLost(),
                     delay, period_packets_sent_, now_ms);
                 period_packets_sent_ = 0;
-                updateEstimate();
+
+                uint64_t max_BW = 0;
+                for (auto &spatial_layer_node : stats_->getNode()["qualityLayers"].getMap()) {
+                  for (auto &temporal_layer_node :
+                      stats_->getNode()["qualityLayers"][spatial_layer_node.first.c_str()].getMap()) {
+                    if (temporal_layer_node.second->value() > max_BW) {
+                      max_BW = temporal_layer_node.second->value() + (temporal_layer_node.second->value() / 100 * 30);
+                    }
+                  }
+                }
+
+                if (max_BW > current_max_bw_set_) {
+                  ELOG_DEBUG("%s message: Updating max bw based on RR: %lu ", connection_->toLog(), max_BW);
+                  current_max_bw_set_ = max_BW;
+                  // We set the max bw based on the max published bitrate plus an headroom of 30%.
+                  // This prevents the bwe estimation to grow at ridiculous levels which causes
+                  // the need of too much time to readjust in case of bw drops.
+                  sender_bwe_->SetMinMaxBitrate(0, max_BW);
+                }
+
+              updateEstimate();
             }
           }
           break;
         case RTCP_PS_Feedback_PT:
           {
-            if (chead->getBlockCount() == RTCP_AFB) {
+            if (chead->getBlockCount() == RTCP_AFB && !ignore_rembs_) {
               char *uniqueId = reinterpret_cast<char*>(&chead->report.rembPacket.uniqueid);
               if (!strncmp(uniqueId, "REMB", 4)) {
                 received_remb_ = true;
                 int64_t now_ms = ClockUtils::timePointToMs(clock_->now());
                 uint64_t bitrate = chead->getBrMantis() << chead->getBrExp();
                 ELOG_DEBUG("%s message: Updating Estimate with REMB, bitrate %lu", connection_->toLog(),
                     bitrate);
+                if (bitrate > current_max_bw_set_) {
+                  ELOG_DEBUG("%s message: Updating max bw based on REMB bitrate : %lu", connection_->toLog(), bitrate);
+                  current_max_bw_set_ = bitrate;
+                  sender_bwe_->SetMinMaxBitrate(0, bitrate);
```

Comment:
Yeah you are totally right, this is an absolute hack. 
I set the max value because if I don't, the bw estimation continues growing up to 1gbit, and at that point, if there's a fluctuation due to packet loss for example, it takes tons of seconds to go down to the accurate level.
I wasn't able to make the google code work correclty.
But in real life testing I think it performed perfectly. When rembs are present btw, the algo don't work and everything's work as you wrote it. 
It's a "temporary fix" for the calls in where rembs are not present. 
If you refuse the PR i totally understand, it's maybe not implemented in the correct way.
Again, if I don't set the max value (which I take on the publisher side), and call 
`
sender_bwe_->UpdateReceiverBlock(chead->getFractionLost(),
  | delay, period_packets_sent_, now_ms);
`
the estimation don't stop to grow